tests/integration/pkg/helpers/api_rule.go

```diff
@@ -123,13 +124,13 @@ func ApplyApiRuleV2Alpha1ExpectError(toExecute RetryableApiRule, onRetry Retryab
 			if err != nil {
 				return err
 			}
-			if apiStatus.Status.State != "Error" {
-				log.Println("APIRule status not Error: " + apiStatus.Status.Description)
-				return errors.New("APIRule status not Error: " + apiStatus.Status.Description)
+			if apiStatus.Status.State != status {
+				log.Printf("APIRule status not %s: %s", status, apiStatus.Status.Description)
+				return fmt.Errorf("APIRule status not %s: %s", status, apiStatus.Status.Description)
```

Comment:
Error should not be capitalized https://go.dev/wiki/CodeReviewComments#error-strings